src/sftp.tf

```
@@ -0,0 +1,75 @@
+# To fully configure the storage account, there are some manual steps. See
+# https://pagopa.atlassian.net/wiki/spaces/DEVOPS/pages/511738518/Enable+SFTP+with+Azure+Storage+Account
+
+resource "azurerm_resource_group" "sftp" {
+  name     = "${local.project}-sftp-rg"
+  location = var.location
+}
+
+module "sftp" { # add private endpoint (dns zone storage account) per uat-prod
+  source = "git::https://github.com/pagopa/azurerm.git//storage_account?ref=v2.18.0"
+
+  name                = replace("${local.project}-sftp", "-", "")
+  resource_group_name = azurerm_resource_group.sftp.name
+  location            = azurerm_resource_group.sftp.location
+
+  account_kind             = "StorageV2"
+  account_tier             = "Standard"
+  account_replication_type = var.sftp_account_replication_type
+  access_tier              = "Hot"
+  is_hns_enabled           = true
+
+  network_rules = {
+    default_action             = "Deny"
+    bypass                     = ["AzureServices"]
+    ip_rules                   = var.sftp_ip_rules
+    virtual_network_subnet_ids = []
+  }
+
+  tags = var.tags
+}
+
+resource "azurerm_private_endpoint" "sftp" {
+  count = var.sftp_enable_private_endpoint ? 1 : 0
+
+  name                = "${module.sftp.name}-endpoint"
+  resource_group_name = azurerm_resource_group.sftp.name
+  location            = azurerm_resource_group.sftp.location
+  subnet_id           = var.sftp_subnet_id
```

```suggestion
  subnet_id           = var.sftp_subnet_id
```
this subnet should be a dedicated private endpoint subnet